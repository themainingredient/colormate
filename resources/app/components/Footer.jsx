import React from 'react';
import styled from 'styled-components';
import Button from './Button';
import Globals from '../Global.styles';
import useHover from '../hooks/useHover';
import { closeWindow } from '../helpers/window.ts';

const { colors, fonts } = Globals;

const FooterWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-left: 24px;
  padding-right: 18px;
  height: 78px;
  width: 320px;
  background-color: ${colors.LightGrey};
`;

const MadeBy = styled.a`
  color: ${colors.MediumGrey};
  font-family: ${fonts.SFPro.reg};
  font-size: 11px;
  text-decoration: none;

  &:hover {
    color: ${colors.TMIBlue};
    cursor: pointer;
  }
`;

const Bold = styled.span`
  font-family: ${fonts.SFPro.bold};
`;

const FeedbackMadeByWrapper = styled.div``;

const Feedback = styled.a`
  font-family: ${fonts.SFPro.bold};
  color: ${colors.DarkGrey};
  font-size: 12px;
  text-decoration: none;

  &:hover {
    color: ${colors.TMIBlue};
  }
`;

const openUrlInBrowser = (url) => {
  window.postMessage('openUrlInBrowser', url);
};

const Footer = () => {
  const [isHovered, hoverRef] = useHover();

  return (
    <FooterWrapper>
      <FeedbackMadeByWrapper>
        <Feedback href='mailto:colormate@themainingredient.co'>
          <span role='img' aria-label='letter-with-heart'>
            💌
          </span>{' '}
          Report & support here
        </Feedback>
        <br />
        <MadeBy onClick={() => openUrlInBrowser('https://www.themainingredient.co')}>
          Made by <Bold>The Main Ingredient</Bold>
        </MadeBy>
      </FeedbackMadeByWrapper>
      <Button ref={hoverRef} type='button' onClick={() => closeWindow()}>
        {isHovered ? '👍' : 'Done'}
      </Button>
    </FooterWrapper>
  );
};

export default Footer;
