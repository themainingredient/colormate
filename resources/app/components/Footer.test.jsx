import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Footer from './Footer';

afterEach(cleanup);

describe('Footer.jsx', () => {
  test('renders and displays correctly', () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
