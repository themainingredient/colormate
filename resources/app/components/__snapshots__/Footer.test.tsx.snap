// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.jsx renders and displays correctly 1`] = `
.c4 {
  height: 37px;
  width: 92px;
  font-family: SFProBold;
  font-size: 14px;
  color: #FFFFFF;
  border-radius: 23px;
  background-color: #4E41FF;
  border: none;
}

.c4:hover {
  background-color: #3B2EEB;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 18px;
  height: 78px;
  width: 100%;
  background-color: #E8E9EF;
}

.c2 {
  color: #B5B8C6;
  font-family: SFProRegular;
  font-size: 11px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  color: #4E41FF;
  cursor: pointer;
}

.c3 {
  font-family: SFProBold;
}

.c1 {
  font-family: SFProBold;
  color: #4d4f59;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  color: #4E41FF;
}

<div
  class="c0"
>
  <div
    class=""
  >
    <a
      class="c1"
      href="mailto:colormate@themainingredient.co"
    >
      <span
        aria-label="letter-with-heart"
        role="img"
      >
        ðŸ’Œ
      </span>
       
      Report & support here
    </a>
    <br />
    <a
      class="c2"
    >
      Made by 
      <span
        class="c3"
      >
        The Main Ingredient
      </span>
    </a>
  </div>
  <button
    class="c4"
    type="button"
  >
    Done
  </button>
</div>
`;
